### PR TITLE
Added invalid email filtering

### LINKED ISSUE #839: Enhance email input field
Can be taken up after https://github.com/bigbinary/neeto-ui/pull/826 is merged.

https://www.loom.com/share/d1872aba07b64581ba494b2a531f91c2

@Thejus-Paul _a

cc: @karthiknmenon @ajmaln

### PR DESCRIPTION
**Pull request by Thejus-Paul**
Fixes #839 
- Added invalid email filtering.
- Added a prop to toggle filter invalid emails.
- Added condition to check for empty emails.

### CODE REVIEW

**Comment by Thejus-Paul** [comment]
filter_invalid_emails

@karthiknmenon _a Please review.

**Comment by karthiknmenon** [comment]
@ajmaln _a Please review.

File: lib/components/EmailInput.js [hunk 1]
```diff
@@ -92,7 +96,18 @@ const EmailInput = ({
...
+            ?
+            <div className="flex">
+              {"Please make sure all emails are valid."}
+              <Typography
+                className="cursor-pointer"
+                style="body3"
+                weight="bold"
+                onClick={handleFilterEmails}
+              >
+                &nbsp;Click here to remove invalid emails.
```
Comment (amaldinesh7) [inline lib/components/EmailInput.js:108]: Let's allow this upon a prop. @karthiknmenon what's your thought on providing the option to customize this text?

File: lib/components/EmailInput.js [hunk 2]
```diff
@@ -38,6 +38,10 @@ const EmailInput = ({
 }) => {
   const [inputValue, setInputValue] = useState("");
 
+  const validEmails = value.filter((email) => email.valid && email);
```
Comment (amaldinesh7) [inline lib/components/EmailInput.js:41]: This is supposed to be a function, not a variable.

**Review by amaldinesh7 — CHANGES_REQUESTED** [review]
@Thejus-Paul _a Please check the comments.

**Comment by Thejus-Paul** [comment]
@amaldinesh7 _a Please review.

File: lib/components/EmailInput.js (hunk 1 above)
Comment (karthiknmenon) [reply lib/components/EmailInput.js]: @amaldinesh7 Yes. I agree. 

cc: @Thejus-Paul

File: lib/components/EmailInput.js [hunk 3]
```diff
@@ -35,11 +36,16 @@ const EmailInput = ({
   onChange,
   error = "",
   onBlur,
+  filterInvalidEmails,
   disabled = false,
   ...otherProps
 }) => {
   const [inputValue, setInputValue] = useState("");
 
+  const validEmails = () => value.filter(email => email.valid && email);
```
Comment (amaldinesh7) [inline lib/components/EmailInput.js:45]: update the name to `getValidEmails` or `renderValidEmails`

**Review by amaldinesh7 — CHANGES_REQUESTED** [review]
@Thejus-Paul _a A minor suggestion, please check.

**Comment by Thejus-Paul** [comment]
@amaldinesh7 _a Please check.

**Pull request merged**